.github/workflows/tests.yml

```diff
@@ -22,7 +22,16 @@ jobs:
         #run: [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14]
     steps:
     - uses: actions/checkout@v4
-    - name: setup python
+    - name: setup python 3.12
+      uses: actions/setup-python@v5
+      with:
+        python-version: 3.12
+    - name: Install coreutils on macOS
+      if: matrix.os == 'macos-latest'
+      run: |
+        brew update
+        brew install coreutils
```

Comment:
why do we need coreutils?

Reply:
There's no nproc by default on Mac so coreutils adds it but now I'm thinking that I could replace the brew code with  something like: 
`echo 'alias nproc="sysctl -n hw.ncpu"' >> $GITHUB_ENV`.  Will try it out later today.

Reply:
I need to investigate this one a bit more. Github doesn't like the alias, and I'm a bit rusty on the Actions CI. 

worse case I add this back and take out the `brew update` I must note that if brew finds itself out of date it will auto-run `brew update`.

Reply:
oh it's not worth getting brew involved just for nproc. let's just get rid of nproc